Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -452,12 +471,19 @@ class TabTrayController: UIViewController, UITabBarDelegate, UICollectionViewDel
     }
 
     private func numberOfColumnsForCurrentSize() -> Int {
-        if self.traitCollection.verticalSizeClass == UIUserInterfaceSizeClass.Compact {
-            return TabTrayControllerUX.NumberOfColumnsRegular
-        } else if self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClass.Compact {
-            return TabTrayControllerUX.NumberOfColumnsCompact
+        let compactLayout = profile.prefs.boolForKey("CompactTabLayout") ?? true
+        let idiom = UIDevice.currentDevice().userInterfaceIdiom
+        let orientation = UIDevice.currentDevice().orientation
+
+        if idiom == .Phone {
```

Comment:
How come we're using the device + orientation instead of this view controller's UITraitCollection's horizontalSizeClass/verticalSizeClass properties to differentiate?